app/models/invoice.rb

```diff
@@ -3,6 +3,19 @@ class Invoice < ApplicationRecord
   belongs_to :merchant
 
   has_many :transactions
+  has_many :invoice_items
 
   validates :status, presence: true
+
+  def total
+    invoice_items.reduce(0) do |total, item|
+      total += (item.unit_price * item.quantity)
+    end
+  end
+
+  def successful?
```

Comment:
@TheKaterTot I think that's exactly right! Let's talk more tomorrow, but I believe that `success` is fundamentally a characteristic of transactions. By proxy, it's (to *some* degree) a characteristic of invoices and invoice_items, but moving it too far away from those models feels strange to me (personal opinion, for sure, and could even see an argument that those last two are too far).

We'll discuss this tomorrow, but you might want to take a look at `.merge`. Essentially, it allows you to scope your join down to some subset of the joined model using a class method on that model.